raiden/network/transport/matrix/utils.py

```diff
@@ -347,11 +366,12 @@ def get_reachability_from_matrix(self, user_ids: Iterable[str]) -> AddressReacha
     def _maybe_address_reachability_changed(self, address: Address) -> None:
         # A Raiden node may have multiple Matrix users, this happens when
         # Raiden roams from a Matrix server to another. This loop goes over all
-        # these users and uses the "best" presence. IOW, if there is a single
+        # these users and uses the "best" presence. IOW, if there is at least one
         # Matrix user that is reachable, then the Raiden node is considered
         # reachable.
         userids = self._address_to_userids[address].copy()
-        composite_presence = {self._userid_to_presence.get(uid) for uid in userids}
+        presence_to_uid = {self._userid_to_presence.get(uid): uid for uid in userids}
+        composite_presence = set(presence_to_uid.keys())
```

Comment:
What is the difference here to the old code?

Reply:
We preserve the mapping `user_id <-> presence` in order to pull the `avatar_url` for **a** `user_id` that has the `new_presence`.

Reply:
The old code was only concerned with ordering presences.